crates/scmi/src/devices/iio.rs

```diff
@@ -0,0 +1,843 @@
+        &self.sensor
+    }
+    fn sensor_mut(&mut self) -> &mut Sensor {
+        &mut self.sensor
+    }
+
+    fn initialize(&mut self) -> Result<(), String> {
+        let mut axes: Vec<Axis> = vec![];
+        match fs::read_dir(&self.path) {
+            Ok(iter) => {
+                for dir_entry in iter {
+                    match dir_entry {
+                        Ok(entry) => self.register_iio_file(entry, &mut axes),
+                        Err(error) => {
+                            return Err(format!(
+                                "Error when reading IIO device path {}: {}",
+                                &self.path, error
+                            ))
+                        }
+                    }
+                }
+            }
+            Err(error) => {
+                return Err(format!(
+                    "Cannot read IIO device path {}: {}",
+                    &self.path, error
+                ))
+            }
+        }
+        if axes.is_empty() {
+            return Err(format!(
+                "No {} channel found in {}",
+                &self.channel, &self.path
+            ));
+        }
+        axes.sort_by(|a1, a2| a1.path.cmp(&a2.path));
+        self.axes = axes;
+        Ok(())
+    }
+
+    fn unit(&self) -> u8 {
+        UNIT_MAPPING
+            .iter()
+            .find(|mapping| mapping.channel == self.channel)
+            .map_or(scmi::SENSOR_UNIT_UNSPECIFIED, |mapping| mapping.unit)
+    }
+
+    fn unit_exponent(&self, axis_index: u32) -> i8 {
+        let axis: &Axis = self.axes.get(axis_index as usize).unwrap();
+        axis.unit_exponent + axis.custom_exponent
+    }
+
+    fn number_of_axes(&self) -> u32 {
+        if self.scalar {
+            0
+        } else {
+            self.axes.len() as u32
+        }
+    }
+
+    fn axis_name_prefix(&self) -> String {
+        let channel = &self.channel;
+        let in_prefix = "in_";
+        let out_prefix = "out_";
+        let name: &str = if channel.starts_with(in_prefix) {
+            channel.strip_prefix(in_prefix).unwrap()
+        } else if channel.starts_with(out_prefix) {
+            channel.strip_prefix(out_prefix).unwrap()
+        } else {
+            &channel[..]
+        };
+        let len = min(name.len(), MAX_SIMPLE_STRING_LENGTH - 1);
+        String::from(&name[..len])
+    }
+
+    fn reading_get(&mut self) -> DeviceResult {
+        let mut result = vec![];
+        for axis in &self.axes {
+            let value = self.read_axis(axis)?;
+            result.push(MessageValue::Unsigned((value & 0xFFFFFFFF) as u32));
+            result.push(MessageValue::Unsigned((value >> 32) as u32));
+            result.push(MessageValue::Unsigned(0));
+            result.push(MessageValue::Unsigned(0));
+        }
+        Result::Ok(result)
+    }
+}
+
+fn read_number_from_file<F: FromStr>(path: &Path) -> Result<Option<F>, ScmiDeviceError> {
+    let exists = path.try_exists();
```

Comment:
Any reason for checking if the file exists first instead of calling read_to_string and matching the `NotFound` error kind? should simplify the code a bit by having one single match

Reply:
No real reason, reworked as you suggest.